File: CodingTest_Inflearn/src/main/java/String/Section1_07.java
package String;

import java.util.Scanner;

// 회문문자열 (앞에 읽거나 뒤에서 읽을 떄 같은 문자열)
public class Section1_07 {
    public String solution(String str) {
        // String answer = "YES";
        // 1. toUpperCase 사용하여 문자열 확인
        /*
        str = str.toUpperCase();
        int len = str.length();
        for(int i=0 ; i<len/2 ; i++) {
            if(str.charAt(i) != str.charAt(len-i-1)) {
                return "NO";
            }
        }
         */
        String answer = "NO";
        String tmp = new StringBuilder(str).reverse().toString();   // 뒤집기
        if(str.equalsIgnoreCase(tmp)) answer = "YES";

        return answer;
    }

    public static void main(String[] args) {
        Section1_07 s = new Section1_07();
        Scanner sc = new Scanner(System.in);
        String str = sc.next();
        System.out.println(s.solution(str));
    }
}

File: CodingTest_Inflearn/src/js/String/Section1_07.js
let str = 'gooG';
//let answer = 'YES';

// 1. toUpperCase 사용하여 문자열 확인
/*
str = str.toUpperCase();
let len = str.length;

for(let i=0; i<len/2 ; i++) {
  if(str[i] != str[len-i-1]) {
    answer = 'NO';
  }
}
console.log(answer);
*/

// 2. reverse 사용
let answer = 'NO';
let tmp = str.split('').reverse().join('');

// Java에서 사용한 equalsIgnoreCase 가 자바스크립에는 지원되지 않으므로 문자열을 소문자로 변경하는 toLowerCase 사용
if(str.toLowerCase(tmp)) answer = 'YES';

console.log(answer);
